### PR TITLE
Use branch 8007-contract_type_aliasing

### LINKED ISSUE #953: Support dbt-core 8592 type aliasing in model contracts
### Description

With pull request https://github.com/dbt-labs/dbt-core/pull/8592 the behavior of types in contracts changed. We need to tweak a test case to match the change in behavior.

### Solution

Update the test case.

### PR DESCRIPTION
**Pull request by gshank**
resolves #953

### Problem

Need to match changes in https://github.com/dbt-labs/dbt-core/pull/8592

### Solution

Fix the test.   Add TEXT => STRING conversion to TYPE_LABELS.

### Checklist

- [x] I have read [the contributing guide](https://github.com/dbt-labs/dbt-core/blob/main/CONTRIBUTING.md) and understand what's expected of me
- [x] I have run this code in development and it appears to resolve the stated issue
- [x] This PR includes tests, or tests are not required/relevant for this PR
- [x] This PR has no interface changes (e.g. macros, cli, logs, json artifacts, config files, adapter interface, etc) or this PR has already received feedback and approval from Product or DX

### CODE REVIEW

File: dbt/adapters/bigquery/column.py [hunk 1]
```diff
@@ -13,12 +13,9 @@
 @dataclass(init=False)
 class BigQueryColumn(Column):
     TYPE_LABELS = {
-        "STRING": "STRING",
```
Comment (colin-rogers-dbt) [inline dbt/adapters/bigquery/column.py:16]: Shouldn't we just be adding `TEXT`? Why delete the other types?

Comment (gshank) [reply dbt/adapters/bigquery/column.py:16]: There is no point to types that are the same on both sides. They do nothing except (in some cases) change the type from lowercase to uppercase.

Comment (gshank) [reply dbt/adapters/bigquery/column.py:16]: I removed them in the dbt-core PR because they were causing unnecessary churning in the test cases.

**Comment by gshank** [comment]
The test is currently failing because in test_constraints.py the "integer" type is being translated to "int64". It looks like the tests were working before with "integer". Is that conversion not necessary?

**Review by colin-rogers-dbt — APPROVED** [review]

**Pull request merged**